### PR TITLE
DDCE-2639 : Upgrade sbt + other libs and run scalafmt + scalastyle

### PR DESCRIPTION
**Pull request by lmugnai**

### CODE REVIEW

File: project/AppDependencies.scala [hunk 1]
```diff
@@ -6,48 +6,47 @@ object AppDependencies {
...
+    "com.vladsch.flexmark"    % "flexmark-all"             % "0.62.2",
+    "org.mockito"             % "mockito-all"              % "1.10.19",
+    "org.pegdown"             % "pegdown"                  % "1.6.0",
+    "com.ironcorelabs"       %% "cats-scalatest"           % "3.1.1"
   ).map(_ % "test, it")
 
-
   private val jettyVersion = "9.2.24.v20180105"
 
   val jettyOverrides = Seq(
```
Comment (frikit) [inline project/AppDependencies.scala:35]: also every time we do updates we were trying to remove these overrides to have less things to maintain.

this was done for ```    "com.github.tomakehurst"  % "wiremock"                 % "2.27.2",```
if you try to upgrade to ```      "com.github.tomakehurst"        % "wiremock-jre8"            % "2.33.2",``` I think we are able to fix the overrides issue with wiremock server and then you can delete all overrides

I tried a small test it was complaining on ```    "com.fasterxml.jackson.module" %% "jackson-module-scala"     % "2.13.3"``` but once you add this seems to be working, anyway I am not able to do full round of test manual, AT... but its worth a try

**Review by frikit — CHANGES_REQUESTED** [review]

File: project/AppDependencies.scala (hunk 1 above)
Comment (lmugnai) [reply project/AppDependencies.scala:35]: I think we are straying way off the scope of the ticket now. This was a simple sbt update which I don't think should turn into a big refactor. Let's create a new ticket for this.

Comment (frikit) [reply project/AppDependencies.scala:35]: I am happy with we will create a new jira for this

**Review by frikit — APPROVED** [review]
LGTM!

**Pull request merged**